src/OpenLoco/Vehicles/VehicleBody.cpp

```diff
@@ -1225,12 +1225,12 @@ namespace OpenLoco::Vehicles
         auto yaw = (sprite_yaw + 16) & 0x3F;
         auto firstBogie = var_38 & Flags38::isReversed ? backBogie : frontBogie;
         auto unkFactor = 5;
-        if (!trackIdToSparkDirection[firstBogie->var_2C / 8])
+        if (!trackIdToSparkDirection[(firstBogie->var_2C.road._data >> 3)])
```

Comment:
This is interesting in that it will access out of bounds memory if unk8 is set. I've no idea what unk8 represents atm and if it is an issue for this.